src/renderer/features/wallets/DerivationsAddressModal/DerivationsAddressModal.tsx

```diff
@@ -0,0 +1,124 @@
+import { DdAddressInfoDecoded, DynamicDerivationRequestInfo } from '@renderer/components/common/QrCode/common/types';
+
+export type ShardedAccountWithShards = ShardedAccount & { shards: ShardAccount[] };
+type DerivationsAccounts = Omit<ShardedAccountWithShards | ChainAccount, 'accountId' | 'walletId' | 'id'>;
+
+const createDerivations = (accounts: DerivationsAccounts[]): DynamicDerivationRequestInfo[] => {
+  return accounts.reduce<DynamicDerivationRequestInfo[]>((acc, account) => {
+    if ('shards' in account) {
+      acc.push(
+        ...(account as ShardedAccountWithShards).shards.map((shard) => ({
+          derivationPath: shard.derivationPath,
+          genesisHash: account.chainId,
+          encryption: shard.cryptoType,
+        })),
+      );
+    } else {
+      acc.push({
+        derivationPath: (account as ChainAccount).derivationPath,
+        genesisHash: account.chainId,
+        encryption: (account as ChainAccount).cryptoType,
+      });
+    }
+
+    return acc;
+  }, []);
+};
+
+type Props = {
+  walletName: string;
+  rootKey: Omit<BaseAccount, 'walletId' | 'id'>;
+  accounts: DerivationsAccounts[];
+  isOpen: boolean;
+  onComplete: () => void;
+  onClose: () => void;
+};
+
+export const DerivationsAddressModal = ({ rootKey, accounts, onClose, isOpen, walletName, onComplete }: Props) => {
+  const { t } = useI18n();
+  const [isScanStep, toggleIsScanStep] = useToggle(false);
+  const derivations = createDerivations(accounts);
+
+  const handleScanResult = (result: DdAddressInfoDecoded[]) => {
+    const derivationsByPath = keyBy(result, (d) => d.derivationPath + d.encryption);
+    createWallet(derivationsByPath);
+  };
+
+  const createDerivedAccounts = (
+    derivedKeys: Dictionary<DdAddressInfoDecoded>,
+  ): Omit<ChainAccount | ShardedAccountWithShards, 'walletId' | 'id'>[] =>
+    accounts.map((account) => {
+      if ('shards' in account) {
+        return {
+          ...account,
+          shards: (account as ShardedAccountWithShards).shards.map((shard) => ({
+            ...shard,
+          })),
+        };
+      } else {
+        return {
+          ...account,
+          accountId: toAccountId(
+            derivedKeys[(account as ChainAccount).derivationPath + (account as ChainAccount).cryptoType].publicKey
+              .public,
+          ),
+        };
+      }
+    });
+
+  const createWallet = (derivedKeys: Dictionary<DdAddressInfoDecoded>) => {
+    const accountsToSave = createDerivedAccounts(derivedKeys);
+
+    walletModel.events.polkadotVaultCreated({
+      wallet: {
+        name: walletName.trim(),
+        type: WalletType.POLKADOT_VAULT,
+        signingType: SigningType.POLKADOT_VAULT,
+      },
+      accounts: accountsToSave,
+      root: rootKey,
+    });
+
+    onComplete();
+  };
+
+  return (
+    <BaseModal
+      isOpen
+      contentClass={isScanStep ? undefined : 'px-5 py-4'}
+      title={t('onboarding.paritySigner.generateAddressesModalTitle')}
+      onClose={() => {}}
```

Comment:
Should we mark it somehow so we don't forget to finish it? Or it will be added in the next PR?

Reply:
Forgot to add it. Fixed